rules/CodingStyle/Rector/MethodCall/PreferThisOrSelfMethodCallRector.php

```diff
@@ -120,10 +120,8 @@ public function configure(array $configuration): void
 
     private function processToSelf(MethodCall | StaticCall $node): ?StaticCall
     {
-        if ($node instanceof StaticCall && ! $this->isNames(
-            $node->class,
-            [ObjectReference::SELF, ObjectReference::STATIC]
-        )) {
+        // class is already "self", let's skip it
+        if ($node instanceof StaticCall && $this->isName($node->class, ObjectReference::SELF)) {
```

Comment:
@samsonasik I have noticed your PR too late, before I sent this one.
The bug seems here, the `self` name should be skipped here.

Reply:
Please chery-pick commit https://github.com/rectorphp/rector-src/pull/2758/commits/cfc30b2a21e1c715a8fd28daf58468c4c6092a7f to ensure the functionality is same, as it seems due to printing, not different of self/this

Reply:
I've started with long code, then shorten it. It could be as short as:

```php
self::asertSame(
	'',
	''
);
```

I've added your commit to have it in CI :+1: